src/plugin/ethTransaction.ts

```diff
@@ -321,11 +332,36 @@ export class EthereumTransaction implements Interfaces.Transaction {
     this.assertHasAction()
 
     if (!this.requiresParentTransaction) {
-      // set gasLimit if not already set, set it using the execution Priority specified for this transaction
-      // NOTE: we don't set fees here if we'll have a parent trnsaction. That will happen when the parent tx is set
-      if (isNullOrEmptyEthereumValue(this._actionHelper.action.gasLimit)) {
+      const { explicitGasPriceValue, explicitGasLimitValue, explicitGasPriceOrGasLimitIsProvided } =
```

Comment:
Having nested ifs makes prepareToBeSigned() now very hard to read and understand
Please factor out all the fee setting code into a seperate function. something like setFeeUsingTxPriorityOrOverride(). Be sure to have a clear function comment for it

Reply:
done